### PR TITLE
feat:Add warning notes to delayed tasks (RWA-772)

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-772

### Change description ###

Add warning notes to delayed tasks

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/WarningTaskWorkerHandlerTest.java [hunk 1]
```diff
@@ -87,17 +112,35 @@ void should_complete_warning_external_task_Service() {
...
+                String warning = warningValues.getValuesAsJson();
+                modifications.put("warningList", DmnValue.dmnStringValue(warning));
+            } catch (JsonProcessingException e) {
+                // do nothing
+            }
+            return new AddProcessVariableRequest(modifications);
         }
 
         @Test
         void should_complete_warning_external_task_Service_with_duplicate_warnings() {
```
Comment (indvamshi) [inline src/test/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/WarningTaskWorkerHandlerTest.java:136]: If possible, can we test with arbitrary values. For eg: warningCode is different, but text is same.

**Review by indvamshi — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/WarningTaskWorkerHandlerTest.java [hunk 2]
```diff
@@ -115,14 +158,92 @@ void should_complete_warning_external_task_Service_with_duplicate_warnings() {
...
+                "hasWarnings", true,
+                "warningList", expectedWarningValues
+            );
+            verify(externalTaskService).complete(externalTask, expectedProcessVariables);
+            verify(taskManagementServiceApi).addTaskNote(S2S_TOKEN, externalTask.getId(), getExpectedWarningRequest());
+            verify(camundaClient).updateProcessVariables(
+                S2S_TOKEN, PROCESS_INSTANCE_ID,
+                getAddProcessVariableRequest(expectedWarningValues)
+            );
         }
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/WarningTaskWorkerHandlerTest.java:239]: should there be a test that shows only a delayed Task will have the addWarningToDelayedProcesses logic applied?

**Review by BenNzewi — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/CamundaProcess.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    private final String id;
+    private final String definitionId;
+    private final String businessKey;
+    private final String caseInstanceId;
+    private final Boolean suspended;
+    private final Boolean ended;
+    private final String tenantId;
+
+    @JsonCreator
+    public CamundaProcess(@JsonProperty("id") String id,
```
Comment (indvamshi) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/CamundaProcess.java:24]: I presume by default its camelCase for Camunda , if it is then we don't ned @JsonProperty to set it explicitly.

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/CamundaProcessVariables.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+
+import static uk.gov.hmcts.reform.waworkflowapi.clients.model.DmnValue.booleanValue;
+import static uk.gov.hmcts.reform.waworkflowapi.clients.model.DmnValue.dmnStringValue;
+
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@EqualsAndHashCode
+@ToString
+public class CamundaProcessVariables {
```
Comment (indvamshi) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/CamundaProcessVariables.java:18]: Can you please add pojo tests for all the new pojos.

**Review by indvamshi — CHANGES_REQUESTED** [review]
LGTM, with minor changes required.

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/CamundaProcess.java (hunk 3 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/CamundaProcess.java]: done

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/CamundaProcessVariables.java (hunk 4 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/model/CamundaProcessVariables.java:18]: added

**Review by BenNzewi — APPROVED** [review]

**Review by indvamshi — APPROVED** [review]

**Pull request merged**